### PR TITLE
Add Back Button and Hide Input Section After Dataset Upload #196

### LINKED ISSUE #196: Add Back Button and Hide Input Section After Dataset Upload
After a dataset is uploaded, the data is displayed on the right side of the screen, but the left part of the screen becomes unresponsive, preventing users from entering new details or interacting with the input section. To improve user experience, the left part should be hidden after uploading the dataset, and only the data table should be displayed. A "Back" button should be provided to allow users to return to the original input screen.

**What Must Be Changed:**
After a dataset is uploaded, hide the input section (left side) and only display the data table (right side).
Introduce a "Back" button that becomes visible after the input section is hidden. This button should be prominently labeled and easily accessible for user navigation. Implement functionality for the "Back" button that, when clicked, restores the original input form on the left side of the screen.
Ensure that the data table remains visible on the right side while the input form is displayed again.

**Expected Outcome:**
Once the dataset is uploaded, the input section will automatically be hidden, and the data table will be the only visible component on the screen. Users will have access to a "Back" button, which allows them to return to the original input form, making the interface more responsive and user-friendly.

**Acceptance Criteria:**

- [ ] Hide Input Section After Upload: Upon dataset upload, the input section on the left must automatically hide, leaving only the data table on the right side visible. The screen should focus solely on displaying the uploaded data.
- [ ] A "Back" button must appear after the input section is hidden, allowing users to navigate back to the original input screen. The button should be clearly labeled and easy to locate.
- [ ] Restore Original View: Clicking the "Back" button must restore the original input form alongside the data table on the right. Ensure smooth transitions between the data table view and the input form view, providing a seamless user experience.

### PR DESCRIPTION
**Pull request by Canada101**
Fixes #196

**What was changed?**

I added the hiding of the input side bar upon entering a dataset and a back button that would appear after inputting data and make the sidebar reappear. 

**Why was it changed?**

This was changed to make the ui flow better and give a more professional look and user experience.

**How was it changed?**

I added a label to the sidebar and a separate back action button that was not visible upon starting the program. this was done in the ui file. When the upload data button is clicked the sidebar gets hidden and the back button appears and when the back button is clicked this brings up the side bar. this was done by adding logic in the server file for when the buttons are clicked.

**How was it tested**

I tested this by running test files and ensuring you could see the back button and that when clicking it would bring back the sidebar. 
[backButtonCapstoneIssue.pdf](https://github.com/user-attachments/files/17359078/backButtonCapstoneIssue.pdf)

**Screenshots that show the changes (if applicable):**

### CODE REVIEW

File: code/ui.R [hunk 1]
```diff
@@ -127,7 +128,13 @@ ui <- navbarPage(
...
         ),
         mainPanel(
-          tags$div(id = "placeholder")
+          tags$div(id = "placeholder"),
+          #logic for back button outside of side pannel
+          actionButton(
+            inputId = "backButton",
+            label = "Back",
+            style = "display: none;"
+          )
```
Comment (srepalli3) [inline code/ui.R:137]: good job! very clear and concise

File: code/server.r [hunk 2]
```diff
@@ -12,6 +12,14 @@ server <- function(input, output, session) {
...
     datasetsUploadedID(TRUE)  # Set the reactive value to TRUE on upload data button click
   })
+  observeEvent(input$uploadData, {
+    shinyjs::hide(id = "sidebar")      # Hide the sidebar content
+    shinyjs::show(id = "backButton")   # Show the "Back" button
+  })
+  observeEvent(input$backButton, {
+    shinyjs::show(id = "sidebar")      # Show the sidebar content
+    shinyjs::hide(id = "backButton")   # Hide the "Back" button
+  })
```
Comment (srepalli3) [inline code/server.r:22]: again, clear and concise

**Review by srepalli3 — APPROVED** [review]

**Comment by kate-holdener** [comment]
I think the "Back" button is confusing and doesn't resolve the issue of the application not being responsive (you can't upload another dataset). Also, once you use the "Back" button, there is no way to hide the left hand menu again. The "Back" button would make sense if it allowed us to get back to the menu and change parameters, but it doesn't allow us to change parameters. 

Once the left hand menu disappears, the output looks strangely offset to the right, with a lot of blank space on the left.
Screenshot 2024-10-15 at 1 15 41 PM


Having the file upload menu disappear makes sense after the upload, I'm not sure if "Back" really makes sense. I've played with the application, and it feels awkward. Also, when switching the dropdown menu to show many entries (100), the back button ends up at the bottom and is hard to find.

@BollepalliPranitha @Canada101 I encourage you to carefully think through the design here to improve usability of the application.

**Comment by Massi-Papi** [comment]
> I think the "Back" button is confusing and doesn't resolve the issue of the application not being responsive (you can't upload another dataset). Also, once you use the "Back" button, there is no way to hide the left hand menu again. The "Back" button would make sense if it allowed us to get back to the menu and change parameters, but it doesn't allow us to change parameters.
> 
> Once the left hand menu disappears, the output looks strangely offset to the right, with a lot of blank space on the left. Screenshot 2024-10-15 at 1 15 41 PM
> 
> Having the file upload menu disappear makes sense after the upload, I'm not sure if "Back" really makes sense. I've played with the application, and it feels awkward. Also, when switching the dropdown menu to show many entries (100), the back button ends up at the bottom and is hard to find.
> 
> @BollepalliPranitha @Canada101 I encourage you to carefully think through the design here to improve usability of the application.

I agree with this. Once you hit the "back" button, you can't really do anything. It just brings you back to the main page, and you aren't able to input anything new. 
It also appears whenever you hit "Upload Data", and it is a bit of a nuisance on the screen. You should have the button hidden until everything loads. A good way to do that is with a boolean: `False = Hidden, True = Showing`
Especially with the reset data button that will incorporated later, this isn't necessary.

**Comment by BollepalliPranitha** [comment]
The primary goal of introducing the "Back" button was to enhance user experience by temporarily hiding the unresponsive input section after a dataset upload, allowing the user to focus on the full display of the data table. The idea was to provide a clear view of the uploaded data without distractions while ensuring users have an accessible way to return to the input form for resetting or updating the dataset if needed.
@Canada101 
However, as noted, the current implementation does not fully address the issue regarding layout and responsiveness. The blank space on the left after the input section is hidden creates an imbalance in the interface, making the output look offset to the right.
The "Back" button currently appears whenever "Upload Data" is clicked, which is visible while the data is still loading. We should only show the "Back" button after the dataset has fully loaded to avoid unnecessary clutter. 
We need to discuss and plan a more effective approach to incorporate these changes while ensuring a seamless user experience.

**Review by BollepalliPranitha — CHANGES_REQUESTED** [review]
The current implementation does not fully address the issue regarding layout and responsiveness. The blank space on the left after the input section is hidden creates an imbalance in the interface, making the output look offset to the right.

**Comment by BollepalliPranitha** [comment]
As this is not working as expected , and being a low priority issue, closing this PR

**Pull request closed without merging**